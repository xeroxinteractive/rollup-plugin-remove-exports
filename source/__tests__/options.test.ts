import stripExports from '../index';
import { rollup } from 'rollup';
import path from 'path';

const originalWarn = console.warn;
const mockWarn = jest.fn();

beforeAll(() => {
  console.warn = mockWarn;
});

afterAll(() => {
  console.warn = originalWarn;
});

test('sourceMap default', async () => {
  const bundle = await rollup({
    input: path.join(__dirname, '__specs__/default-named.js'),
    treeshake: false,
    plugins: [stripExports({})],
  });
  const { output } = await bundle.generate({ format: 'es', sourcemap: true });
  expect(output[0].map).toBeTruthy();
});

test('sourceMap true', async () => {
  const bundle = await rollup({
    input: path.join(__dirname, '__specs__/default-named.js'),
    treeshake: false,
    plugins: [stripExports({ sourceMap: true })],
  });
  const { output } = await bundle.generate({ format: 'es', sourcemap: true });
  expect(output[0].map).toMatchSnapshot();
});

test('sourceMap false', async () => {
  const bundle = await rollup({
    input: path.join(__dirname, '__specs__/default-named.js'),
    treeshake: false,
    plugins: [stripExports({ sourceMap: false })],
  });
  const { output } = await bundle.generate({ format: 'es', sourcemap: true });
  expect(output[0].map?.mappings).toBe(';;;;;;;;');
  expect(mockWarn).toHaveBeenCalledWith(
    "Sourcemap is likely to be incorrect: a plugin (strip-exports) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help"
  );
});
