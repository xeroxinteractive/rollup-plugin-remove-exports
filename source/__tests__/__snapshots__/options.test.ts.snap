// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sourceMap true 1`] = `
SourceMap {
  "file": "default-named.js",
  "mappings": "AAAA,sBAA6B;AAC7B,gBAAuB;AACvB,oBAA2B;AAC3B,wBAA+B;AAC/B,iCAAwC;AACxC;AACA;;",
  "names": Array [],
  "sources": Array [
    "source/__tests__/__specs__/default-named.js",
  ],
  "sourcesContent": Array [
    "export const test = 'string';
export const test1 = 5;
export const test3 = false;
export const test4 = undefined;
export const test5 = new Symbol('duck');

export default (foo) => {
  console.log(foo + 'bar');
};
",
  ],
  "version": 3,
}
`;
